### ISSUE TITLE
Bug: [require-await] Misleading location for arrow functions

### ISSUE THREAD

**Issue by reduckted** [typescript-eslint/typescript-eslint#9606]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=MYewdgzgLgBAhgJwSA7jAvPCBPMwYAUAlBgHwwDeAsAFAwwICmUArgmANy0C%2BXNtcHHhgAzFnigBLcKPHBilWvSat2fbrSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aRAR1ko9oEMA7v0r50UYdCaRwYAL4g5QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const arrow = async () => {
                    // ~~ Async arrow function 'arrow' has no 'await' expression.
    return;
};
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/require-await": "warn"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The location of the warning should be on the part of the code that needs to change to fix the problem, like this:

```ts
const arrow = async () => {
           // ~~~~~ Async arrow function 'arrow' has no 'await' expression.
    return;
};
```

### Actual Result

The location of the warning is on the `=>`, which means it's not obvious what the problem is.

### Additional Info

Ideally, the location of the warning would be on the `async` keyword because you fix this problem by removing the `async` keyword.

A full function (as opposed to an arrow function) suffers from a similar problem, although the location is _marginally_ better because it covers the entire function header, which includes the `async` keyword (as can be seen in the playground link).

**Comment by kirkwaiblinger** [comment]
Hi @reduckted!

I agree with you completely! But, this is an extension rule, so this behavior would need to be addressed upstream with eslint. Which, in this case, I already unsuccessfully proposed, see https://github.com/eslint/eslint/issues/18339.

Closing this, since we won't take action unless the upstream situation changes. 🙃
